Update release checklists post 0.55.2 (#8118)
numba/numba: .github/ISSUE_TEMPLATE/first_rc_checklist.md

* [ ] Merge change log changes.
    - [ ] "PR with changelog entries".
* [ ] Create X.Y release branch.

> * [ ] Dependency version pinning on release branch

Reviewer: ```suggestion
* [ ] Commit build farm configuration and tag with release version.
* [ ] Dependency version pinning on release branch
```
Suggest that the above is added as a reminder.

Author: This has been added for the final release checklist. I believe this should suffice. Tagging the buildfarm for a first RC isn't that useful as often changes need to be made for the final!

  * [ ] Pin llvmlite to `>=0.A.0rc1,<0.A+1.0`.
  * [ ] Pin NumPy if needed

>   * [ ] Pin tbb if needed

Reviewer: ```suggestion
  * [ ] Pin TBB if needed
```

Author: sure

* [ ] Annotated tag X.Y.Zrc1 on release branch (no `v` prefix).
* [ ] Build and upload conda packages on buildfarm (check "upload").
* [ ] Build wheels and sdist on the buildfarm (check "upload").
* [ ] Verify packages uploaded to Anaconda Cloud and move to `numba/label/main`.
* [ ] Upload wheels and sdist to PyPI (upload from `ci_artifacts`).
* [ ] Verify wheels for all platforms arrived on PyPi.
* [ ] Initialize and verify ReadTheDocs build.
* [ ] Send RC announcement email / post announcement to discourse group.
* [ ] Post link to Twitter.

### Post Release:

> * [ ] Clean up `ci_artifacts` by moving files to subdirectories

Reviewer: ```suggestion
* [ ] Clean up `ci_artifacts` by moving files to sub-directories
```
